### DISCUSSION TITLE
Prisma utility type Args<Type, Operation> does not work as expected.

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#22137] [Q&A]
### Question

I am using the utility type Prisma.Args<Type,Operation> however type
```
{
  data: Prisma.Args<Type,Operation>['data']
}
```
is not assignable to type Prisma.Args<Type,Operation> and I dont know why. Can you please help me? I am digging this issue since 3 days. Help would mean a lot. Thank you! 

here is also a screenshot of some tests I conducted
<img width="1174" alt="Screenshot 2023-11-27 at 20 11 14" src="https://github.com/prisma/prisma/assets/23662270/ddea5d75-7bed-41bb-9002-4188a2c5a062">
this.setActionParameter function input parameter is of type Prisma.Args<PrismaService[M], F> in this case F is equal to 'create'
if I set the input parameter to type Prisma.TypeMap['model'][M]['operations'][F]['args'] typescript accepts the type.
As far as I know the Prisma.Args utilities returns exatcly this Prisma.TypeMap['model'][M]['operations'][F]['args'], then why the type is not recognized?
here is the error from typescript:
```
**Argument of type** 
{ **data:** (Prisma.Without<Prisma.vehicleCreateInput, Prisma.vehicleUncheckedCreateInput> & Prisma.vehicleUncheckedCreateInput) | (Prisma.Without<Prisma.vehicleUncheckedCreateInput, Prisma.vehicleCreateInput> & Prisma.vehicleCreateInput) | (Prisma.Without<Prisma.makeCreateInput, Prisma.makeUncheckedCreateInput> & Prisma.makeUncheckedCreateInput) | (Prisma.Without<Prisma.makeUncheckedCreateInput, Prisma.makeCreateInput> & Prisma.makeCreateInput) | (Prisma.Without<Prisma.modelCreateInput, Prisma.modelUncheckedCreateInput> & Prisma.modelUncheckedCreateInput) | (Prisma.Without<Prisma.modelUncheckedCreateInput, Prisma.modelCreateInput> & Prisma.modelCreateInput) | (Prisma.Without<Prisma.fuel_typeCreateInput, Prisma.fuel_typeUncheckedCreateInput> & Prisma.fuel_typeUncheckedCreateInput) | (Prisma.Without<Prisma.fuel_typeUncheckedCreateInput, Prisma.fuel_typeCreateInput> & Prisma.fuel_typeCreateInput) | (Prisma.Without<Prisma.body_typeCreateInput, Prisma.body_typeUncheckedCreateInput> & Prisma.body_typeUncheckedCreateInput) | (Prisma.Without<Prisma.body_typeUncheckedCreateInput, Prisma.body_typeCreateInput> & Prisma.body_typeCreateInput) | (Prisma.Without<Prisma.gearbox_typeCreateInput, Prisma.gearbox_typeUncheckedCreateInput> & Prisma.gearbox_typeUncheckedCreateInput) | (Prisma.Without<Prisma.gearbox_typeUncheckedCreateInput, Prisma.gearbox_typeCreateInput> & Prisma.gearbox_typeCreateInput) | (Prisma.Without<Prisma.engine_typeCreateInput, Prisma.engine_typeUncheckedCreateInput> & Prisma.engine_typeUncheckedCreateInput) | (Prisma.Without<Prisma.engine_typeUncheckedCreateInput, Prisma.engine_typeCreateInput> & Prisma.engine_typeCreateInput); }
**is not assignable to parameter of type** ' Args<PrismaService[M], "create">'.ts(2345)
```

If you more information please feel free to ask i will provide.

**Comment by SevInf** [comment]
Hi @vtonchev.
Could you please provide reproducible example we can run and see the error happening? I am not 100% clear on what you want to achive and your report missing some necessary information:
- your schema
- prisma version
- exact definitions of `getCreateDtoParam` and `setActionParameters` functions and `PrismaService` class and `M` type

But the best thing would be minimal github repo we can checkout and run.

For what it worth, this snippet works correctly for me:

```ts
type DataType<M extends keyof typeof prisma> = Prisma.Args<(typeof prisma)[M], 'create'>['data']

async function createUser(data: DataType<'user'>) {
  await prisma.user.create({ data })
}
```

`User` is defined as follows:

```prisma
model User {
  id String @id @default(uuid())
}
```

What's the difference between this and your case is hard to tell without all of that information.

We'd also appreciate if the code you want us to try out will be shared as copy-pasteble text rather than screenshot.
Thank you!

**Reply by ludralph** [reply]
Hey @vtonchev ,

Did you get a chance to see SevInf response? It would be helpful if you could provide the information requested. If this is no longer relevant, you can go ahead and close the discussion.
